File: LeetCode Solutions/Recursion I/Merge Two Sorted Lists.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* mergeTwoLists(ListNode* l1, ListNode* l2) {
        ListNode *newNode = new ListNode(0);
        ListNode *requiredNode = newNode;
        while(l1 != NULL || l2 != NULL){
            if(l2 == NULL){
                requiredNode->next = l1;
                break;
            }
            if(l1 == NULL){
                requiredNode->next = l2;
                break;
            }
            if(l1->val > l2->val){
                requiredNode->next = new ListNode(l2->val);
                l2 = l2->next;
            } else{
                requiredNode->next = new ListNode(l1->val);
                l1 = l1->next;
            }
            requiredNode = requiredNode->next;
        }
        return newNode->next;
    }
};

File: LeetCode Solutions/LeetCode 75/Level 1/Day 3 Linked List/Merge Two Sorted Lists.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode mergeTwoLists(ListNode list1, ListNode list2) {
        ListNode newNode = new ListNode(0);
        ListNode requiredNode = newNode;
        while(list1 != null || list2 != null){
            if(list1 == null){
                requiredNode.next = list2;
                break;
            }
            if(list2 == null){
                requiredNode.next = list1;
                break;
            }
            if(list1.val > list2.val){
                requiredNode.next = new ListNode(list2.val);
                list2 = list2.next;
            } else{
                requiredNode.next = new ListNode(list1.val);
                list1 = list1.next;
            }
            requiredNode = requiredNode.next;
        }
        return newNode.next;
    }
}